### PR TITLE
Update tools/expreiment.py

### PR DESCRIPTION
**Pull request by eunwoosh**
### Summary

This PR updates tools/expreiment.py as below.

- [x] Make experiment directory name same always
- [x] Parse all metric values in `otx eval` output file.
- [x] Fix minor unit test bug

#### Why is experiment name different in each run before?
When parsing variables in YAML recipe file, keys of variables are saved in a set.
Using the set, not only which values to use but also order of dict (python 3.6+ keeps order in default dict) is decided.
In contrast to my expectation, `set` doesn't have always same order.
So, I added a code to sort the set, which makes constant order in every runs.

#### How is `otx eval` ouptut parsed now?
After this PR is merged, all metrics in `otx eval` output file are parsed and saved in final experiment result file.
For example, if we assume that `eval` is executed after `train`, `export` and `optimize` in detection task,
metrics as below are saved

test_score => f-measure(train)
Recall(train) is newly added.
Precision(train) is newly added.
export_model_score => f-measure(export)
Recall(export) is newly added.
Precision(export) is newly added.
optimize_model_score => f-measure(optimize)
Recall(optimize) is newly added.
Precision(optimize) is newly added.
And other metrics are same as previous

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by goodsong81 — DISMISSED** [review]
LGTM

File: tests/unit/algorithms/segmentation/adapters/mmseg/datasets/pipelines/test_transforms.py [hunk 1]
```diff
@@ -110,11 +110,11 @@ class TestNormalize:
     @pytest.mark.parametrize(
         "mean,std,to_rgb,expected",
         [
-            (1.0, 1.0, True, np.array([[[1.0, 0.0, 0.0]]], dtype=np.float32)),
-            (1.0, 1.0, False, np.array([[[-1.0, 0.0, 0.0]]], dtype=np.float32)),
+            ([1.0 for _ in range(3)], [1.0 for _ in range(3)], True, np.array([[[1.0, 0.0, -1.0]]], dtype=np.float32)),
+            ([1.0 for _ in range(3)], [1.0 for _ in range(3)], False, np.array([[[-1.0, 0.0, 1.0]]], dtype=np.float32)),
```
Comment (goodsong81) [inline tests/unit/algorithms/segmentation/adapters/mmseg/datasets/pipelines/test_transforms.py:114]: Nitpick: [1.0, 1.0, 1.0] is better than [1.0 for _ in range(3)] for readability. But it's all OK.

**Review by goodsong81 — APPROVED** [review]
Thanks for the fix.

File: tests/unit/algorithms/segmentation/adapters/mmseg/datasets/pipelines/test_transforms.py (hunk 1 above)
Comment (sungmanc) [reply tests/unit/algorithms/segmentation/adapters/mmseg/datasets/pipelines/test_transforms.py:114]: I also prefer [1.0, 1.0, 1.0]

**Review by sungmanc — APPROVED** [review]

File: tests/unit/algorithms/segmentation/adapters/mmseg/datasets/pipelines/test_transforms.py (hunk 1 above)
Comment (eunwoosh) [reply tests/unit/algorithms/segmentation/adapters/mmseg/datasets/pipelines/test_transforms.py:114]: I agree with you. I'll use that format next time.

**Pull request merged**